tests/cases/fourslash/overloadOnConstCallSignature.ts

```diff
@@ -11,8 +11,8 @@
 
 goTo.marker('1');
 verify.signatureHelpCountIs(4);
-verify.currentSignatureHelpIs('foo(name: string): string');
+verify.currentSignatureHelpIs('foo(name: \'order\'): string');

```

Comment:
I personally feel like this updated behaviour is actually better, since it indicates to the author immediately that there are specialised signatures to pick from.

In any case, I am not sure that the extra complexity of doing something different would be worth the cost (it might be, if the only cost is to add a loop to skip `hasStringLiteral` signatures in this situation), but that is just my opinion.